xl_auth/permission/forms.py

```diff
@@ -112,6 +116,8 @@ def set_defaults(self, permission, new_user_id=None):
         self.registrant.default = permission.registrant
         self.cataloger.default = permission.cataloger
         self.cataloging_admin.default = permission.cataloging_admin
+        if self.global_registrant:
```

Comment:
What happens here if `not current_user.is_admin`, given that `self.global_registrant` is deleted from `self` in the super class if it is not? (I would assume that since self.global_registrant is missing, it refers to the field definition on the *class*, which I would expect to be truthy?)

Reply:
It will be `None`. 

Removing the `if`:
```    
File "/home/olov/kod/xl_auth/xl_auth/permission/forms.py", line 120, in set_defaults
    self.global_registrant.default = permission.global_registrant
    AttributeError: 'NoneType' object has no attribute 'default'

```

This is because how the WTForms `Form` base class handles delete.
site-packages/wtforms/fomr.py
```
    def __delitem__(self, name):
        del self._fields[name]
        setattr(self, name, None)

    def __delattr__(self, name):
        if name in self._fields:
            self.__delitem__(name)
        else:
            # This is done for idempotency, if we have a name which is a field,
            # we want to mask it by setting the value to None.
            unbound_field = getattr(self.__class__, name, None)
            if unbound_field is not None and hasattr(unbound_field, '_formfield'):
                setattr(self, name, None)
            else:
                super(Form, self).__delattr__(name)
```

See also 
https://wtforms.readthedocs.io/en/latest/specific_problems.html#removing-fields-per-instance

Reply:
Aha! Then it is the idiomatic way to do it using WTForms. Thanks for explaining!